app/scripts/services/api.js

```diff
@@ -191,8 +199,11 @@ angular.module('openshiftConsole')
         AuthService.withUser();
         return;
       }
-      // Otherwise go to the error page, the server might be down.
-      Navigate.toErrorPage("Unable to load details about the server. If the problem continues, please contact your system administrator.", "API_DISCOVERY", true);
+      // Otherwise go to the error page, the server might be down.  Can't use Navigate.toErrorPage or it will create a circular dependency
+      $window.location.href = URI('error').query({
```

Comment:
in order to use APIService.kindtoResource in navigate.js I had to switch to not using Navigate.toErrorPage, otherwise there was a circular dependency.  This seemed like the better of the two options.